### PR TITLE
[chore] Replaces multierr usage with go native errors package

### PR DESCRIPTION
**Pull request by dylanlott**
## Description

The recent upgrade to Go 1.20 enabled the errors package `Join` function to replace usage of multierr `Combine` and `Append` functions. Please note it cannot be removed as a dependency from our `go.mod` because it is an indirect dependency.

## Issue

Follow up changes from review of #861 and merging #910  

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Replaces `multierr.Combine` and `multierr.Append` usage with `errors.Join`

## Testing

- [x] `make develop_test`; if any code changes were made
- [ ] `make test_e2e` on [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any code changes were made
- [ ] `e2e-devnet-test` passes tests on [DevNet](https://pocketnetwork.notion.site/How-to-DevNet-ff1598f27efe44c09f34e2aa0051f0dd); if any code was changed
- [ ] [Docker Compose LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md); if any major functionality was changed or introduced
- [ ] [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any infrastructure or configuration changes were made

## Required Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added, or updated, [`godoc` format comments](https://go.dev/blog/godoc) on touched members (see: [tip.golang.org/doc/comment](https://tip.golang.org/doc/comment))
- [ ] I have tested my changes using the available tooling
- [ ] I have updated the corresponding CHANGELOG

### If Applicable Checklist

- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -197,6 +196,7 @@ require (
 	go.uber.org/atomic v1.10.0 // indirect
 	go.uber.org/dig v1.15.0 // indirect
 	go.uber.org/fx v1.18.2 // indirect
+	go.uber.org/multierr v1.9.0 // indirect
```
Comment (Olshansk) [inline go.mod:199]: Can we remove this dependency altogether? (go.sum) should be updated too

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (h5law) [reply go.mod:199]: I assume another import from uber uses the multierr as a dep so we cannot get rid of it altogether until they update

File: shared/node.go [hunk 2]
```diff
@@ -186,12 +186,12 @@ func (node *Node) handleEvent(message *messaging.PocketEnvelope) error {
 		err_p2p := node.GetBus().GetP2PModule().HandleEvent(message.Content)
 		err_ibc := node.GetBus().GetIBCModule().HandleEvent(message.Content)
 		// TODO: Remove this lib once we move to Go 1.2
```
Comment (h5law) [inline shared/node.go:188]:
```suggestion
```

File: shared/node.go [hunk 3]
```diff
@@ -186,12 +186,12 @@ func (node *Node) handleEvent(message *messaging.PocketEnvelope) error {
...
-		return multierr.Combine(err_p2p, err_ibc)
+		return errors.Join(err_p2p, err_ibc)
 	case messaging.StateMachineTransitionEventType:
 		err_consensus := node.GetBus().GetConsensusModule().HandleEvent(message.Content)
 		err_p2p := node.GetBus().GetP2PModule().HandleEvent(message.Content)
 		// TODO: Remove this lib once we move to Go 1.2
```
Comment (h5law) [inline shared/node.go:193]:
```suggestion
```

**Review by h5law — APPROVED** [review]
LGTM, left a couple comments on lines we can remove but changes are simple enough to :shipit:

File: go.mod (hunk 1 above)
Comment (Olshansk) [reply go.mod:199]: @dylanlott Can you verify?

File: shared/node.go (hunk 3 above)
Comment (Olshansk) [reply shared/node.go:193]: 👍

File: go.mod (hunk 1 above)
Comment (dylanlott) [reply go.mod:199]: It's being imported by some other dependency of ours, but there are no more usages of multierr in our specific code that I could find.

Comment (dylanlott) [reply go.mod:199]: > Can we remove this dependency altogether? (go.sum) should be updated too

I actually don't think go.sum needs to change, the dependency was only made indirect, and the version stayed the same, so I suspect the dependency graph stayed the same 🤔

**Review by Olshansk — APPROVED** [review]

**Pull request merged**